src/structui/ui.py

```
@@ -287,12 +287,22 @@ def render_primitive_input(k, v, parent_node):
                 def make_on_change(prop_key=k, prop_type=meta.get('type')):
                     def handler(e):
                         val = e.value
+                        is_original_int = isinstance(v, int) and not isinstance(v, bool)
+                        is_original_float = isinstance(v, float)
+
                         if prop_type == 'integer' and val is not None and val != '':
                             try: val = int(float(val))
```

Ruff (0.15.13)

[error] Multiple statements on one line (colon)

(E701)